pkg/sql/bulkingest/split.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package bulkingest
+
+import (
+	"context"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+)
+
+func splitAndScatterSpans(ctx context.Context, db *kv.DB, spans []roachpb.Span) error {
+	// TODO(jeffswenson): when distributing work, we should re-scatter ranges if
+	// there are a small number of nodes with remaining work.
+	for _, span := range spans {
+		expirationTime := db.Clock().Now().Add(time.Hour.Nanoseconds(), 0)
+		if err := db.AdminSplit(ctx, span.Key, expirationTime); err != nil {
+			return err
+		}
+	}
+	for _, span := range spans {
+		req := &kvpb.AdminScatterRequest{
+			// Randomly distributed ranges to make ingest more balanced.
```

Comment:
More as a TODO for giant tables we may want larger spans? Even with big ranges a 20TB table would be a lot of splits. We can probably do wider spans in that case

Reply:
The output of merge and import will be range sized SSTs. split_picker.go creates splits so that each SST will end up in its own range. So a 20 TiB table would end up with 20 TiB / 128 MiB per SST ~= 8k ranges